src/Pagination/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { Request } from 'express'
+
+export const DEFAULT_LIMIT = 10000 // let's use the default as the max as well
+
+export const getPaginationParams = (req: Request) => {
+  const { limit, page } = req.query
+  const parsedLimit = Number(limit)
+  const parsedPage = Number(page)
+  return {
+    limit:
+      limit && !isNaN(parsedLimit) && parsedLimit < DEFAULT_LIMIT
+        ? parsedLimit
+        : undefined,
+    page: page && !isNaN(parsedPage) ? parsedPage : undefined,
+  }
```

Comment:
I think it might be useful to use `parseInt` here? I see the following cases being "better" handled:

```ts
Number('') // 0
parseInt('', 10) // NaN

Number('.0') // 0
parseInt('.0', 10) // NaN

Number('1.3') // 1.3
parseInt('1.3', 10) // 1
```

This would also allow the condition below to only use `!isNaN(parsedLimit)`, unless you don't want to allow `0` in which case I'd write:

```ts
  const { query } = req
  const limit = parseInt(query.limit, 10)
  const page = parseInt(query.page, 10)
  return {
    limit:
      !isNaN(limit) && limit > 0 && limit < DEFAULT_LIMIT
        ? limit
        : undefined,
    page: !isNaN(page) && page > 0 ? page : undefined,
  }
```


What do you think?